Carbon.Cryptography/Functions/Find-Certificate.ps1

```diff
@@ -0,0 +1,366 @@
+      subject's common name and the certificate's subject alternate names.
+    * `LiteralHostName`: return certificates that authenticate the given hostname. Matches against the subject's common
+      name and the certificate's subject alternate names.
+    * `KeyUsageName`: return certificates that have the given enhanced key usage, searching each usage's friendly name.
+    * `KeyUsageOId`: return certificates that have the given enhanced key usage, searching each usage's object ID.
+    * `Trusted`: return certificates that are trusted/verified.
+
+    You can search in the local machine or current user accounts (not both) by passing the account to the
+    `StoreLocation` parameter. You can search in different stores by passing the store name to the `StoreName`
+    parameter.
+
+    .EXAMPLE
+    Find-CCertificate -Active -HostName 'dev.example.com' -KeyUsageName 'Server Authentication' -Trusted -HasPrivateKey
+
+    Demonstrates how to search for a certificate using multiple criteria. In this example, we're looking for a TLS
+    certificate that can be used with the `dev.example.com` hostname.
+    #>
+    [CmdletBinding()]
+    param(
+        [String] $Subject,
+
+        [String] $LiteralSubject,
+
+        [switch] $Active,
+
+        [switch] $HasPrivateKey,
+
+        [String] $HostName,
+
+        [String] $LiteralHostName,
+
+        [String] $KeyUsageName,
+
+        [String] $KeyUsageOid,
+
+        [switch] $Trusted,
+
+        [Security.Cryptography.X509Certificates.StoreLocation] $StoreLocation,
+
+        [Security.Cryptography.X509Certificates.StoreName] $StoreName =
+            [Security.Cryptography.X509Certificates.StoreName]::My
+    )
+
+    Set-StrictMode -Version 'Latest'
+    Use-CallerPreference -Cmdlet $PSCmdlet -Session $ExecutionContext.SessionState
+
+    if( $Subject )
+    {
+        Write-Verbose ("Subject        like  $($Subject)")
+    }
+    if( $LiteralSubject )
+    {
+        Write-Verbose ("Subject        eq    $($LiteralSubject)")
+    }
+    if( $Active )
+    {
+        Write-Verbose ('Active         True')
+    }
+    if( $HasPrivateKey )
+    {
+        Write-Verbose ('HasPrivateKey  True')
+    }
+    if( $HostName )
+    {
+        Write-Verbose ("HostName       like  $($HostName)")
+    }
+    if( $LiteralHostName )
+    {
+        Write-Verbose ("HostName       eq    $($LiteralHostName)")
+    }
+    if( $KeyUsageName )
+    {
+        Write-Verbose ("Key Usage      $($KeyUsageName)")
+    }
+    if( $KeyUsageOid )
+    {
+        Write-Verbose ("Key Usage OID  $($KeyUsageOid)")
+    }
+    if( $Trusted )
+    {
+        Write-Verbose ("Trusted        True")
+    }
+    if( $StoreLocation )
+    {
+        Write-Verbose ("StoreLocation  $($StoreLocation)")
+    }
+    if( $StoreName )
+    {
+        Write-Verbose ("StoreName      $($StoreName)")
+    }
+    Write-Verbose ''
+
+    function Test-Object 

```

Comment:
This function seems worthy of a few tests. Maybe add a TODO to refactor out into its own file and add tests.